core/pta/ocall.c

```diff
@@ -0,0 +1,344 @@
+// SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2020, Microsoft Corporation
+ */
+
+#include <kernel/pseudo_ta.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <optee_rpc_cmd.h>
+#include <pta_ocall.h>
+#include <tee/uuid.h>
+
+#define PTA_NAME "ocall.ta"
+
+#define PTR_ADD(ptr, offs) ((void *)((uintptr_t)(ptr) + (uintptr_t)(offs)))
+
+static TEE_Result compute_required_mobj_size(uint32_t param_types,
+					     TEE_Param params[TEE_NUM_PARAMS],
+					     size_t *required_size)
+{
+	TEE_Param *param;
```

Comment:
generic comment: always initialize local variables where defined. See [coding-standards](https://optee.readthedocs.io/en/latest/general/coding_standards.html#coding-standards), 3rd bullet.

Reply:
Interesting. This requirement cuts both ways though: always initializing variables precludes the compiler from warning the developer when the variable is used in a location where its value should have been updated prior to the usage. I've run into a few of those in fact while writing this patch set.

From my days of working on Windows, we'd leave uninitialized variables which are expected to be filled later on. That way, if our control flow was busted and a variable was used without having been set to the expected value, the compiler would bark at us. And we compile with warnings as errors, so we'd fail the build altogether.

I'll follow the coding standards, but with reservations :) .

Reply:
The rationale is that we want all variables to have a well defined value, i.e., to have the developer think about it and come up with a good value instead of having what often is a random value otherwise (static's etc are an exception). The reason why we've introduced this is because we've been bitten by a couple of security issues in the past due to uninitialized variables. As anything else, there are pros and cons.